src/Google.Cloud.Functions.Framework/CloudEventAdapterTData.cs

```diff
@@ -28,24 +28,27 @@ public sealed class CloudEventAdapter<TData> : IHttpFunction where TData : class
         private readonly CloudEventFormatter _formatter;
         private readonly ILogger _logger;
 
+        /// <summary>
+        /// Constructor without a <see cref="CloudEventFormatter"/>, whose sole purpose is to throw a clear exception
+        /// if an event formatter has not been configured.
+        /// </summary>
+        [Obsolete("This constructor only exists to enable a clear error message. It always throws an exception.", error: true)]
```

Comment:
Nice making it obsolete!

Reply:
And "hard obsolete" at that - it's impossible to call other than via reflection.
We may want to add some more attributes to avoid it being shown by Intellisense - although I don't expect anyone to be creating this by hand anyway.